src/main/java/net/dv8tion/jda/internal/entities/EntityBuilder.java

```diff
@@ -457,6 +457,15 @@ public GuildChannel createGuildChannel(GuildImpl guildObj, DataObject channelDat
             return null;
         }
     }
+    
+    public User.PrimaryGuild createPrimaryGuild(DataObject obj)
+    {
+        long guildId = obj.getLong("identity_guild_id", -1);
+        if (guildId == -1) // Possibly null according to docs
```

Comment:
Instead of using `-1`, which is a valid snowflake, let's just check `obj.isNull("identity_guild_id")`